src/client/services/conversation/conversation.js

```diff
@@ -89,6 +92,46 @@ var ConversationService = SparkBase.extend(
     this.encryptionDisabled = false;
   },
 
+  getExifData: function getExifData(file, buf) {
+    return new Promise(function(resolve) {
+      if (file && file.image && file.mimeType === 'image/jpeg') {
+        new ExifImage({ image : buf }, function (error, exifData) {
+          if (error) {
+            this.logger.info('conversation: error with image rotation ', error.message);
+          }
+          else {
+            if (exifData) {
+              file.image.orientation = exifData.image.Orientation;
+            }
+          }
+          resolve(buf);
+        });
+      }
+      else {
+        resolve(buf);
+      }
+    }.bind(this))
+  },
+
+  fixImageOrientation: function fixImageOrientation(file) {
```

Comment:
Web-client needs to call it once the file share button is clicked, so that the thumbnail generated by the web-client is properly rotated before sent to the server. This is the requirement for all clients so that they remain consistent with the image display everywhere.